spectre_analyses/tasks.py

```diff
@@ -0,0 +1,159 @@
+"""divik task definition for the purpose of use through Celery
+
+Copyright 2018 Spectre Team
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+from contextlib import contextmanager
+from functools import partial
+import json
+import os
+import pickle
+import re
+import shutil
+import sys
+
+import celery
+from celery.utils.log import get_task_logger
+
+import spdata.reader
+import spdata.types as ty
+
+from spectre_analyses.celery import app
+import matlab_hooks as mh
+
+
+FILESYSTEM_ROOT = os.path.abspath(os.sep)
+DATA_ROOT = os.path.join(FILESYSTEM_ROOT, 'data')
+STATUS_PATHS = {
+    'all': FILESYSTEM_ROOT,
+    'done': DATA_ROOT,
+    'processing': os.path.join(FILESYSTEM_ROOT, 'temp'),
+    'failed': os.path.join(FILESYSTEM_ROOT, 'failed')
+}
+Name = str
+Path = str
+DISALLOWED_CHARACTERS = "[^a-zA-Z0-9_-]"
```

Comment:
These are truly disallowed, right? What characters do we accept, if not alphanumeric then?

Reply:
There is `^` at the beginning, which negates content. We support only `a-zA-Z0-9_-`.